File: 0078-subsets/0078-subsets.go

import "slices"
func subsets(nums []int) (ans [][]int) {
    n := len(nums)
    path := []int{}
    var dfs func(int)
    dfs = func(i int) {
        ans = append(ans, slices.Clone(path))
        for j := i; j < n; j++ {
            path = append(path, nums[j])
            dfs(j + 1)
            path = path[:len(path) - 1]
        }
    }
    dfs(0)
    return ans
}

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        int n = nums.size();
        vector<vector<int>> ans;
        vector<int> path;
        auto dfs = [&](this auto&& dfs, int i) -> void {
            ans.emplace_back(path);
            for (int j = i; j < n; j++) {
                path.push_back(nums[j]);
                dfs(j + 1);
                path.pop_back();
            }
        };
        dfs(0);
        return ans;
    }
};